dask/async.py

```diff
@@ -426,69 +426,69 @@ def get_async(apply_async, num_workers, dsk, result, cache=None,
         result_flat = set([result])
     results = set(result_flat)
 
-    dsk = dsk.copy()
-    for f in start_cbs:
-        f(dsk)
-
-    dsk, dependencies = cull(dsk, list(results))
+    try:
+        dsk = dsk.copy()
+        for f in start_cbs:
+            f(dsk)
```

Comment:
Small chances were what got us into this mess (the computation in the issue was a very fast and repeated one to spend more time in these sections).  We could also make a `started_cbs` list.